src/main/java/io/jenkins/plugins/okhttp/api/internals/JenkinsProxyAuthenticator.java

```diff
@@ -47,4 +48,18 @@ public Request authenticate(@Nullable Route route, Response response) throws IOE
 
         return null;
     }
+
+    private boolean isAuthenticationSchemeSupported(@NonNull final String proxyAuthenticateHeader) {
+        final String lowerCaseHeader = proxyAuthenticateHeader.toLowerCase();
+
+        /* According https://square.github.io/okhttp/4.x/okhttp/okhttp3/-authenticator/ :
+
+             | OkHttp will call authenticate with a fake HTTP/1.1 407 Proxy Authentication Required response that has a
+             | Proxy-Authenticate: OkHttp-Preemptive challenge. The proxy authenticator may return either an
+             | authenticated request, or null to connect without authentication.
+
+           That is why we check for the OkHttp-Preemptive header.
+         */
+        return lowerCaseHeader.startsWith("basic") || lowerCaseHeader.equalsIgnoreCase("okhttp-preemptive");
```

Comment:
with the toLowerCase() above, is the equalsIgnoreCase really needed here ? Is not equals enough ?

Reply:
Fixed